testing/upgradetest/versions_test.go

```diff
@@ -249,33 +248,177 @@ func TestGetUpgradableVersions(t *testing.T) {
 }
 
 func TestPreviousMinor(t *testing.T) {
-	currentParsed, err := version.ParseVersion(bversion.Agent)
-	require.NoError(t, err)
+	testCases := map[string]struct {
```

Comment:
can you sort test based on current version?
i'd expect us to test
major release - 9.0.0
first patch of new major - 9.0.1
first minor release 9.1.0
first patch of first minor 9.1.1
xth patch 9.1.15
xth major 9.2.0
first patch 9.2.1 
xth patch 9.2.15

with having 8.x in history of releases.
this should put clear contract and expectations for desired behavior, then once we agree on this we can tweak implemenation to fit